recipes/renode-cli/helpers/renode_build_with_dotnet.ps1

```diff
@@ -0,0 +1,114 @@
+$ErrorActionPreference = "Stop"
+
+# Get logical processors count
+$cpuCount = (Get-CimInstance Win32_Processor).NumberOfLogicalProcessors
+
+# Set environment variables (using robust expansion)
+$SRC_DIR = & cmd.exe /c echo %SRC_DIR%
+$PREFIX = & cmd.exe /c echo %PREFIX%
+$PKG_NAME = & cmd.exe /c echo %PKG_NAME%
+
+# Check for empty environment variables
+if ([string]::IsNullOrEmpty($SRC_DIR)) { throw "SRC_DIR is empty" }
+if ([string]::IsNullOrEmpty($PREFIX)) { throw "PREFIX is empty" }
+if ([string]::IsNullOrEmpty($PKG_NAME)) { throw "PKG_NAME is empty" }
+
+# Set environment variables
+$env:PATH = "${env:BUILD_PREFIX}/Library/mingw-w64/bin;${env:BUILD_PREFIX}/Library/bin;${env:PREFIX}/Library/bin;${env:PREFIX}/bin;${env:PATH}"
+
+# Function to install tests (defined before use in PowerShell)
+function install_tests {
```

Comment:
Similar to my previous comment, I don't see the need to define functions in this script given that everything is just evaluated imperatively here anyway.

Reply:
Done. I don't quite understand, though, I usually use functions to clearly define a scope, like a list of parameters that are needed and also an easy way to add/remove part of the script. In the editor I can also easily collapse them and not be bothered by the boundaries when they are in one big blob